src/poetry/core/packages/specification.py

```diff
@@ -131,6 +131,7 @@ def __hash__(self) -> int:
             ^ hash(self._source_url)
             ^ hash(self._source_reference)
             ^ hash(self._source_resolved_reference)
+            ^ hash(self._source_subdirectory)
```

Comment:
We should add a test to check that two specs with different subdirectories has differently.

Reply:
Added the test for package equality